crates/sui-core/src/authority/epoch_start_configuration.rs

```diff
@@ -47,19 +47,23 @@ pub enum EpochFlag {
     PerEpochFinalizedTransactions,
     ObjectLockSplitTables,
     WritebackCacheEnabled,
+    StateAccumulatorV2Enabled,
 }
 
 impl EpochFlag {
     pub fn default_flags_for_new_epoch(config: &NodeConfig) -> Vec<Self> {
-        Self::default_flags_impl(&config.execution_cache)
+        Self::default_flags_impl(&config.execution_cache, config.state_accumulator_v2)
     }
 
     /// For situations in which there is no config available (e.g. setting up a downloaded snapshot).
     pub fn default_for_no_config() -> Vec<Self> {
-        Self::default_flags_impl(&Default::default())
+        Self::default_flags_impl(&Default::default(), false)
```

Comment:
should this be true? if we download a snapshot we probably want to start with the v2 accumulator after restore

Reply:
Yeah I tagged you in a review comment above asking exactly this. Thanks for confirming here though

Reply:
Actually thought more about this -- if we set to true, then this would enable v2 on snapshot restore irrespective of node config specification... which would break rollout control